### ISSUE TITLE
CoqIDE: If files without .v extension are loaded, the user get's a "coqtop died badly"

### ISSUE THREAD

**Issue by MSoegtropIMC** [coq/coq#10654]
If one loads a file without .v extension in CoqIDE, the user get's a `coqtop died badly error`. This is not terribly helpful.

This error was reported by @kristinas.

**Comment by Zimmi48** [comment]
On what system and which version? I could not reproduce with either Coq 8.9, 8.10, and master.

**Comment by kristinas** [comment]
The 8.10 beta2 custom build produced by Michael that includes the HoTT
package (64 bit version).

**Comment by Zimmi48** [comment]
So it would seem that this is a Windows-specific bug, unless someone else can reproduce on another platform.

**Comment by MSoegtropIMC** [comment]
I did some more testing. The bug is not there in 8.9.1. It is there in an about 2 weeks old build from 8.10 master (the build artifact from my HoTT PR #10588 from July 27th).

If the bug cannot be reproduced on Linux with 8.10 master, it indeed seems to be a Windows specific bug. I will take care of it.

@Zimmi48 : can you please test #10653 on Linux on 8.10 master?

**Comment by Zimmi48** [comment]
I actually tested this on slightly outdated version (8.10+beta1) and an outdated version of master. I will update and try again. I did reproduce #10653 even in 8.9.1 however (more of a usability problem than a bug).

**Comment by SkySkimmer** [comment]
May be related to https://github.com/coq/coq/pull/10557

**Comment by Zimmi48** [comment]
Could not reproduce even on 8.10+beta2 (which does not include #10557 though) and on an up-to-date copy of master.

**Comment by MSoegtropIMC** [comment]
@Zimmi48 : thanks for testing - I guess this is a Windows specific problem then and I will take care of it.

**Comment by Alizter** [comment]
Here is a way to reproduce similar issues. Here is my version when coqide starts up:
```
Welcome to CoqIDE, an Integrated Development Environment for Coq
You are running The Coq Proof Assistant, version localhost:/home/ali/coq,master (f508ddcd2cfff152b8d6291d96e4b87ef9fe2ff9)
```
If I go File > Open and then select any file that is not of the form *.v, coqide crashes. I cannot get the coqtop died badly error anymore since I udpated.

The expected behavior should be a warning saying coqide could not load file, rather than crashing the whole IDE.

I am on Ubuntu bionic.

**Comment by Zimmi48** [comment]
@Alizter Is it really any file, or is it limited to filenames that are not valid Coq identifiers? I personally can only reproduce a crash on filenames containing a dash and ending with an extension e.g. `foo-bar.txt` (cf. #10988).

**Comment by Alizter** [comment]
It doesn't crash for file names with no symbols. But it seems to crash if there is an extra `.`  or `-` as you say. Even when it doesn't crash there is no message or anything.

**Comment by Zimmi48** [comment]
A message should appear briefly in the bottom bar. It is easy to miss it. @MSoegtropIMC You confirm that on Windows, even a well-formed Coq identifier without the ending `.v` makes CoqIDE crash?

**Comment by Alizter** [comment]
@Zimmi48 You are correct a message does appear. I did not see this at all.

Opening the wrong file extension is fine, it does give an error at the bottom. However if I open a file like .v.d it crashes. It says `Invalid character '.' in identifier "Basics.v".` when I tried to open `Basics.v.o`.

I suspect this is the same issue with the dash.

**Comment by Zimmi48** [comment]
Indeed, and this was supposed to be fixed in #10557, but apparently it hasn't been fully fixed yet.
